### PR TITLE
Apply CommentThread vscode Plugin API

### PR DESCRIPTION
**Pull request by vinokurig**
#### What it does

Apply [CommentThread vscode Plugin API](https://code.visualstudio.com/api/references/vscode-api#comments)

related CQ: http://dev.eclipse.org/ipzilla/show_bug.cgi?id=22902

fixes https://github.com/eclipse-theia/theia/issues/8492
#### How to test

1. Remove `"@theia/git": "^1.8.0"` from `examples/browser/package.json` file.
2. Download [vscode-git-1.49.3](https://open-vsx.org/api/vscode/git/1.49.3/file/vscode.git-1.49.3.vsix) and copy it to the `plugins` folder.
3. Download [vscode GitHub Pull request plugin](https://github.com/microsoft/vscode-pull-request-github/releases/download/0.20.0/vscode-pull-request-github-0.20.0.vsix) and copy it to the `plugins` folder.
4. Apply GitHub authentication provider:
* Git clone the [github authentication plugin](https://github.com/vinokurig/github-authentication-plugin.git) to the `plugins` folder.
* Paste [your GitHub token](https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token) to the plugin [here](https://github.com/vinokurig/github-authentication-plugin/blob/89b951d1a4280aee62635f5fa2b99feaf92aeacb/src/github-authentication-plugin.ts#L26).
* Compile the plugin. Note that errors might appear in the compilation log but it doesn't prevent the plugin from working properly.
5. Start Theia and  open a project cloned from GitHub with a Pull-Request e.g. https://github.com/vinokurig/environment-variable.git
6. Sign in the `GitHub Pull-Requests Plugin`:
screenshot-serverg0cevid1-che-dev-server-3010 192 168 99 253 nip io-2020 09 23-11_42_23
7. Open a pull-request in the plugin: 
screenshot-localhost_3000-2020 12 16-12_13_45
8. Open a changed file with comments: 
screenshot-localhost_3000-2020 12 16-12_16_21

See: Pull-Request comments are available in the diff. You can add some new comments by clicking the white line and delete and edit existing comments.
#### Review checklist

- [ ] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

**Review by azatsarynnyy — COMMENTED** [review]
When I'm trying to add a comment to the PR, it's failed with the error notification.
And another input area has appeared below after clicking Save button.
Peek 2020-12-16 13-09

**Review by azatsarynnyy — COMMENTED** [review]
After clicking Cancel, it shows the comment (x2) but they're not on the GitHub https://github.com/vinokurig/environment-variable/pull/12

image

**Review by azatsarynnyy — COMMENTED** [review]
I've tried `Delete Comment` button, but it does nothing.
Maybe some additional steps are required to configure the feature on my side?

**Comment by vinokurig** [comment]
@azatsarynnyy
> When I'm trying to add a comment to the PR, it's failed with the error notification.
And another input area has appeared below after clicking Save button.

I couldn't reproduce this issue, I even tried it with another github user:
video-convert (1)
May be something wrong was with GitHub?

>I've tried Delete Comment button, but it does nothing.
Maybe some additional steps are required to configure the feature on my side?

Probably you don't have rights to delete the comment, I have fixed it by limiting the comment actions by access rights.

**Review by RomanNikitenko — COMMENTED** [review]
I tested the changes - I'm able to create, edit and delete a comment - in general it works well for me.
But I faced with the issue that comments were displayed twice, please see a short video

gitgub_PR_issue

@vinokurig 
could you check/test this use case on your side?

**Comment by azatsarynnyy** [comment]
@vinokurig my issues were related to not enough permissions for my token :smile: 
image
Thanks @RomanNikitenko for the advice.
I'll re-check it with the updated token's permissions.

**Review by azatsarynnyy — COMMENTED** [review]
I've verified the base functionality:
- add/edit/delete a comment
- start/finish/delete a review

It works fine so far. Additionally, I want to check several more cases...

**Comment by vinokurig** [comment]
@RomanNikitenko 
> But I faced with the issue that comments were displayed twice

Thank you, fixed it in my latest commit

**Comment by azatsarynnyy** [comment]
I've verified that the comments UI duplication is fixed now. Thanks!

**Comment by azatsarynnyy** [comment]
There's another difference comparing to VS Code.
The comment input hides on focus out. Then it's displayed again once the focus is coming back:

Peek 2020-12-18 16-13

**Comment by azatsarynnyy** [comment]
In VS Code, the input area hides on focus out, only when it's empty.

**Comment by vinokurig** [comment]
@azatsarynnyy 
> In VS Code, the input area hides on focus out, only when it's empty.

Fixed.

**Review by azatsarynnyy — APPROVED** [review]
Thank you @vinokurig for fixing all the issues found during the review.
Everything works well now. :+1:

File: packages/plugin-ext/src/common/plugin-protocol.ts [hunk 1]
```diff
@@ -663,6 +665,18 @@ export interface Keybinding {
     win?: string;
 }
 
+/**
+ * Keybinding contribution
+ */
+export interface Keybinding {
```
Comment (RomanNikitenko) [inline packages/plugin-ext/src/common/plugin-protocol.ts:671]: @vinokurig 
I guess the interface was added by mistake, could you [take a look](https://github.com/eclipse-theia/theia/blob/master/packages/plugin-ext/src/common/plugin-protocol.ts#L702-L721) 
I created a PR to remove it https://github.com/eclipse-theia/theia/pull/9374

**Pull request merged**